MoreLinq/Reactive/Subject.cs

```diff
@@ -0,0 +1,92 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 Atif Aziz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Reactive
+{
+    using System;
+    using System.Collections.Generic;
+    using Delegate = Delegating.Delegate;
+
+    sealed class Subject<T> : IObservable<T>, IObserver<T>
```

Comment:
I can't say I'm thrilled at having yet another Subject (and IObservable helpers). I don't see a way out without depending on Rx though.

Reply:
But then again, this _is_ the way out. :) Adding a dependency on Rx, especially on someone who doesn't use the experimental version, seems rude.